refactor(linter/js-plugins): use `u32` for IDs

overlookmotel opened the pull request:
Follow-on after #12160.

Use `u32` for inner type of `ExternalPluginId` and `ExternalRuleId`. The only advantage of `NonMaxU32` is that `Option<NonMaxU32>` is 4 bytes. That's why we use it for `ScopeId` etc. But `ExternalPluginId` and `ExternalRuleId` aren't stored in `Option`s, so it's simpler to use plain `u32`.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12243?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#12244** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12244?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12243** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12243?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12243?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#12242** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12242?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12241** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12241?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12243).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented:
### Merge activity

* **Jul 12, 8:08 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(linter/js-plugins): use `u32` for IDs (#12243)

Follow-on after #12160.

Use `u32` for inner type of `ExternalPluginId` and `ExternalRuleId`. The only advantage of `NonMaxU32` is that `Option<NonMaxU32>` is 4 bytes. That's why we use it for `ScopeId` etc. But `ExternalPluginId` and `ExternalRuleId` aren't stored in `Option`s, so it's simpler to use plain `u32`.
